### DISCUSSION TITLE
git branch no longer showing in zsh terminal

### DISCUSSION THREAD

**Discussion by Duncan-Wood** [ohmyzsh/ohmyzsh#12266] [Help] [answered]
Prior to the update, if I was in a git repository I could see the branch name in parentheses in my terminal. This was helpful to always ensure I was always on the right branch. Now it doesn't show up anymore.

**Comment by shimamizu** [comment]
Yes, the latest update broke this. Maybe from this update? 
 - 083cc2c [async]      Implement async prompt API and apply to git prompt (#12257)

**Comment by Duncan-Wood** [comment]
I added this to my .zshrc file as a temporary fix. It looks similar but I liked the old look more.

```
function parse_git_branch() {
    git branch 2> /dev/null | sed -n -e 's/^\* \(.*\)/[\1]/p'
}

COLOR_DEF=$'%f'
COLOR_DIR=$'%F{39}'
COLOR_GIT=$'%F{197}'
setopt PROMPT_SUBST
export PROMPT='${COLOR_DIR}%1~ ${COLOR_GIT}$(parse_git_branch)${COLOR_DEF} '
```

**Reply by carlosala** [reply]
@Duncan-Wood doing `omz update` should fix your issue 👍🏻

**Reply by shimamizu** [reply]
thanks for rolling back that :)

**Reply by Duncan-Wood** [reply]
thanks Carlosala!!

**Comment by mcornella** [comment, marked as answer]
Thank you everybody for the fast replies! We found the culprit and sent in the fix. However by default async git prompt is turned off, as this feature is still very early stages.

Please check out https://github.com/ohmyzsh/ohmyzsh/issues/12267#issuecomment-1986926931 for how to check it out and give us your feedback!
